qiskit/circuit/library/arithmetic/piecewise_chebyshev.py

```diff
@@ -377,7 +377,7 @@ class PiecewiseChebyshevGate(Gate):
             from qiskit import QuantumCircuit
             from qiskit.circuit.library.arithmetic import PiecewiseChebyshevGate
 
-            f_x, num_state_qubits, degree, breakpoints = lambda x: np.arcsin(1 / x), 2, 2, [2, 4]
+            f_x, num_state_qubits, degree, breakpoints = lambda x: np.sin(1 / x), 2, 2, [2, 4]
```

Comment:
What prompted this change?

Reply:
With these parameters `PiecewiseChebyshevGate` called the lambda function with values outside the valid domain of `arcsin`, which resulted with a Sphnix warning. I thought it would just be easier to change the function to clean up the warning.